src/app.js

```diff
@@ -1,15 +1,21 @@
-import React from 'react';
+import React, { useState } from 'react';
 
 import { Location } from './contexts/location';
 import ExplorerUI from './components/explorer-ui';
 
-const homeUrl = process.env.TOP_LEVEL;
-
 const App = () => {
-  return (
-    <Location homeUrl={homeUrl}>
+  const [ landingUrl, setLandingUrl ] = useState(new URL(document.location.href).searchParams.get('location'));
+  const [ inputURL, setInputURL ] = useState('');
+
+  return landingUrl ? (
+    <Location landingUrl={landingUrl}>
       <ExplorerUI />
     </Location>
+  ) : (
+    <div>
+      <input name="beginning-url" type="text" value={inputURL} onChange={(e) => setInputURL(e.target.value)} />
+      <button onClick={() => {setLandingUrl(inputURL)}}>Explore</button>
+    </div>
```

Comment:
I think this is fine for the prototype but it seems like something which could be added _into_ the app so that a user could change it without refreshing

Reply:
...and now it is with these latest commits! You can remove part of the url and enter- and it updates the response and the UI.